results/cmd/api/main.go

```diff
@@ -81,3 +96,23 @@ func (s *server) CreateTaskRun(ctx context.Context, req *pb.CreateTaskRunRequest
 	}
 	return taskrunFromClient, nil
 }
+
+func (s *server) DeleteTaskRun(ctx context.Context, req *pb.DeleteTaskRunRequest) (*empty.Empty, error) {
+	database := s.db
+	statement, err := database.Prepare("DELETE FROM taskrun WHERE uid = ?")
+	if err != nil {
+		log.Fatalf("failed to create delete statement: %v", err)
+	}
+	results, err := statement.Exec(req.GetUid())
+	if err != nil {
+		log.Fatalf("failed to execute delete statement: %v", err)
+	}
+	affect, err := results.RowsAffected()
+	if err != nil {
+		log.Fatalf("failed to retrieve results: %v", err)
+	}
+	if affect == 0 {
+		return &(empty.Empty{}), errors.New("row not found: did not find a row with specified condition in database")
```

Comment:
```suggestion
		return nil, status.Errorf(status.NotFound, "TaskRun not found")
```

Use https://pkg.go.dev/google.golang.org/grpc/status?tab=doc instead.

Also in general, I like to avoid database specific terms in user facing error messages, because the implementation of the API server is a detail users don't know/care about. e.g. we could reimplement this using a filesystem instead of a database, but the results API would still be the same!

Reply:
done